dataprofiler/profilers/profiler_options.py

```diff
@@ -346,6 +347,32 @@ def is_prop_enabled(self, prop: str) -> bool:
             is_enabled = option_prop.is_enabled
         return is_enabled
 
+    @classmethod
+    def load_from_dict(
+        cls,
+        data,
+        options: dict | None = None,
+    ) -> BaseOption:
```

Comment:
If this is inheriting BaseOption, why do we need this. also, need to fix the use of `profile` here and above in `BaseOption`